lib/updater.js

```diff
@@ -106,10 +106,10 @@ function _handleCfChunk(cf_chunk, cf_arg, cf_cb) {
          * with the given VM.
          */
         cf_arg.server = { server_uuid: vm.server_uuid };
-        if (vm.state === 'running') {
+        if (vm.state !== 'destroyed') {
```

Comment:
This should be `vm.state === 'running' || vm.state === 'stopped'`. We don't want to add provisioning, failed, invalid (etc) vms to the cache.

As written, the final `else` of this statement is unreachable.